### PR TITLE
refactor: exact instances of some Option/Result/other combinators

### PR DESCRIPTION
**Pull request by huitseeker**
Tool-aided by [comby-rust](https://github.com/huitseeker/comby-rust).

### CODE REVIEW

**Comment by huitseeker** [comment]
@dignifiedquire the intentional commit of this PR is the [first](https://github.com/n0-computer/iroh/pull/466/commits/a9a83c803e1c013017afc61431c0c916c3f45408), the other one being a standard run of clippy auto-fixes. 

The idea is to simplify the code from verbose pattern-matches to idiomatic standard library combinators they are an exact reimplementation of. My opinions on this are loosely held - if you're not convinced of those idioms being simpler, in part of as a group, I'm happy to amend the PR to your desire, or close it altogether.

File: iroh-metrics/src/lib.rs [hunk 1]
```diff
@@ -104,11 +104,7 @@ async fn init_metrics(cfg: Config) -> Option<JoinHandle<()>> {
 /// Initialize the tracing subsystem.
 fn init_tracer(cfg: Config) -> Result<(), Box<dyn std::error::Error>> {
     #[cfg(feature = "tokio-console")]
-    let console_subscriber = if cfg.tokio_console {
-        Some(console_subscriber::spawn())
-    } else {
-        None
-    };
+    let console_subscriber = (cfg.tokio_console).then_some(console_subscriber::spawn());
```
Comment (matthiasbeyer) [inline iroh-metrics/src/lib.rs:107]:
```suggestion
    let console_subscriber = cfg.tokio_console.then(console_subscriber::spawn);
```

In the code right now, `console_subscriber::spawn()` is always called, which is probably not what you want, right?

Also, the parentheses around `cfg.tokio_console` are not required!

File: iroh-p2p/src/node.rs [hunk 2]
```diff
@@ -152,11 +152,7 @@ impl<KeyStorage: Storage> Node<KeyStorage> {
     pub async fn run(&mut self) -> anyhow::Result<()> {
         info!("Local Peer ID: {}", self.swarm.local_peer_id());
 
-        let mut nice_interval = if self.use_dht {
-            Some(tokio::time::interval(NICE_INTERVAL))
-        } else {
-            None
-        };
+        let mut nice_interval = (self.use_dht).then_some(tokio::time::interval(NICE_INTERVAL));
```
Comment (matthiasbeyer) [inline iroh-p2p/src/node.rs:155]:
```suggestion
        let mut nice_interval = self.use_dht.then(|| tokio::time::interval(NICE_INTERVAL));
```

File: stores/flatfs/src/flatfs.rs [hunk 3]
```diff
@@ -173,41 +173,23 @@ impl Flatfs {
...
 
         self.walk().filter_map(move |r| match r {
-            Ok(entry) => {
-                if entry.path().is_file() {
-                    Some(read_file(entry.path()))
-                } else {
-                    None
-                }
-            }
+            Ok(entry) => (entry.path().is_file()).then_some(read_file(entry.path())),
```
Comment (matthiasbeyer) [inline stores/flatfs/src/flatfs.rs:176]:
```suggestion
            Ok(entry) => entry.path().is_file().then(|| read_file(entry.path())),
```

File: stores/flatfs/src/flatfs.rs [hunk 4]
```diff
@@ -173,41 +173,23 @@ impl Flatfs {
...
     pub fn keys(&self) -> impl Iterator<Item = Result<String>> {
         self.walk().filter_map(move |r| match r {
-            Ok(entry) => {
-                if entry.path().is_file() {
-                    Some(key_from_path(entry.path()))
-                } else {
-                    None
-                }
-            }
+            Ok(entry) => (entry.path().is_file()).then_some(key_from_path(entry.path())),
```
Comment (matthiasbeyer) [inline stores/flatfs/src/flatfs.rs:184]:
```suggestion
            Ok(entry) => entry.path().is_file().then(|| key_from_path(entry.path())),
```

File: stores/flatfs/src/flatfs.rs [hunk 5]
```diff
@@ -173,41 +173,23 @@ impl Flatfs {
...
     pub fn stats(&self) -> impl Iterator<Item = Result<KvStats>> {
         self.walk().filter_map(move |r| match r {
-            Ok(entry) => {
-                if entry.path().is_file() {
-                    Some(KvStats::from_path(entry.path()))
-                } else {
-                    None
-                }
-            }
+            Ok(entry) => (entry.path().is_file()).then_some(KvStats::from_path(entry.path())),
```
Comment (matthiasbeyer) [inline stores/flatfs/src/flatfs.rs:192]:
```suggestion
            Ok(entry) => entry.path().is_file().then(|| ̣̣̣KvStats::from_path(entry.path())),
```

File: stores/flatfs/src/flatfs.rs [hunk 6]
```diff
@@ -216,11 +198,7 @@ impl Flatfs {
     pub fn values(&self) -> impl Iterator<Item = Result<Vec<u8>>> {
         self.walk().filter_map(move |r| match r {
             Ok(entry) => {
-                if entry.path().is_file() {
-                    Some(fs::read(entry.path()).map_err(Into::into))
-                } else {
-                    None
-                }
+                (entry.path().is_file()).then_some(fs::read(entry.path()).map_err(Into::into))
```
Comment (matthiasbeyer) [inline stores/flatfs/src/flatfs.rs:201]:
```suggestion
                entry.path().is_file().then(|| fs::read(entry.path()).map_err(Into::into))
```

**Review by matthiasbeyer — CHANGES_REQUESTED** [review]

**Comment by huitseeker** [comment]
Thanks for the review @matthiasbeyer! This makes total sense, I'll also amend [the tooling](https://github.com/huitseeker/comby-rust/issues/1) with your comments!

**Review by dignifiedquire — DISMISSED** [review]

File: iroh-metrics/src/lib.rs [hunk 7]
```diff
@@ -104,11 +104,7 @@ async fn init_metrics(cfg: Config) -> Option<JoinHandle<()>> {
 /// Initialize the tracing subsystem.
 fn init_tracer(cfg: Config) -> Result<(), Box<dyn std::error::Error>> {
     #[cfg(feature = "tokio-console")]
-    let console_subscriber = if cfg.tokio_console {
-        Some(console_subscriber::spawn())
-    } else {
-        None
-    };
+    let console_subscriber = cfg.tokio_console.then(|| console_subscriber::spawn());
```
Comment (b5) [inline iroh-metrics/src/lib.rs:107]: one last bit to placate clippy:
```suggestion
    let console_subscriber = cfg.tokio_console.then(console_subscriber::spawn);
```

**Comment by b5** [comment]
thanks @huitseeker!

**Pull request merged**